### PR TITLE
Settings: rework wallet page

### LINKED ISSUE #2647: Settings: too much text in Settings > Wallet
We should remove buttons with repeated text and add some nice icons (similiar to Settings > Node page):
image

Current Settings > Node page:
image

Current Settings > Wallet page:
image

See also issue #2652

### PR DESCRIPTION
**Pull request by tobtoht**
closes #2647

gui_settings_wallet_1

### CODE REVIEW

**Comment by rating89us** [comment]
If you have time, take a look at #2475

**Comment by selsta** [comment]
@tobtoht Is this still WIP? Also can you close #2634 if this replaces it?

**Comment by selsta** [comment]
Please rebase.

File: components/SettingsListItem.qml [hunk 1]
```diff
@@ -1,65 +1,117 @@
...
+        color: "transparent"
 
-    RowLayout {
-        Layout.fillWidth: true
-        spacing: 0
-
-        ColumnLayout {
+        Rectangle {
+            id: divider
             Layout.fillWidth: true
```
Comment (selsta) [inline components/SettingsListItem.qml]:
```suggestion
```

File: components/SettingsListItem.qml [hunk 2]
```diff
@@ -1,65 +1,117 @@
...
+                width: parent.width - (icon.width + icon.anchors.leftMargin + anchors.leftMargin)
             }
         }
 
-        MoneroComponents.StandardButton {
-            id: button
-            small: true
+        Rectangle {
+            id: bottomDivider
+            Layout.fillWidth: true
```
Comment (selsta) [inline components/SettingsListItem.qml]:
```suggestion
```

File: pages/settings/SettingsWallet.qml [hunk 3]
```diff
@@ -47,20 +48,75 @@ Rectangle {
...
 
             onClicked: appWindow.showWizard()
         }
 
         MoneroComponents.SettingsListItem {
-            buttonText: qsTr("Create wallet") + translationManager.emptyString
-            description: qsTr("Creates a new wallet that can only view and initiate transactions, but requires a spendable wallet to sign transactions before sending.") + translationManager.emptyString
+            iconText: FontAwesome.lock
+            title: qsTr("Lock this wallet") + translationManager.emptyString
+            description: qsTr("Enter your password to unlock it.") + translationManager.emptyString
```
Comment (rating89us) [inline pages/settings/SettingsWallet.qml]:
```suggestion
            description: qsTr("Locks you wallet and asks for the password to unlock it.") + translationManager.emptyString
```

File: pages/settings/SettingsWallet.qml [hunk 4]
```diff
@@ -80,19 +136,9 @@ Rectangle {
...
-                Utils.showSeedPage();
-            }
-        }
-
-        MoneroComponents.SettingsListItem {
-            buttonText: qsTr("Rescan") + translationManager.emptyString
-            description: qsTr("Use this feature if you think the shown balance is not accurate.") + translationManager.emptyString
+            iconText: FontAwesome.repeat
             title: qsTr("Rescan wallet balance") + translationManager.emptyString
+            description: qsTr("Use this feature if you think the shown balance is not accurate.") + translationManager.emptyString
```
Comment (rating89us) [inline pages/settings/SettingsWallet.qml]:
```suggestion
            description: qsTr("Rescans the blockchain, updating and refreshing your balance.") + translationManager.emptyString
```

File: pages/settings/SettingsWallet.qml [hunk 5]
```diff
@@ -47,20 +48,75 @@ Rectangle {
...
+                passwordDialog.onRejectedCallback = function() { appWindow.showWizard(); }
+                if (inputDialogVisible) inputDialog.close()
+                passwordDialog.open();
+            }
+        }
+
+        MoneroComponents.SettingsListItem {
+            iconText: FontAwesome.ellipsisH
+            title: qsTr("Change wallet password") + translationManager.emptyString
+            description: qsTr("Change the password of your wallet.") + translationManager.emptyString
```
Comment (rating89us) [inline pages/settings/SettingsWallet.qml]:
```suggestion
            description: qsTr("Changes the password of your wallet file.") + translationManager.emptyString
```

File: pages/settings/SettingsWallet.qml [hunk 6]
```diff
@@ -47,20 +48,75 @@ Rectangle {
...
+                }
+                passwordDialog.onRejectedCallback = null;
+                passwordDialog.open()
+            }
+        }
+
+        MoneroComponents.SettingsListItem {
+            iconText: FontAwesome.key
+            title: qsTr("Show seed & keys") + translationManager.emptyString
+            description: qsTr("Store this information safely to recover your wallet in the future.") + translationManager.emptyString
```
Comment (rating89us) [inline pages/settings/SettingsWallet.qml]:
```suggestion
            description: qsTr("Displays all the information needed to backup your wallet and safely recover it in the future.") + translationManager.emptyString
```

File: pages/settings/SettingsWallet.qml [hunk 7]
```diff
@@ -47,20 +48,75 @@ Rectangle {
...
+                    }
+                }
+                passwordDialog.onRejectedCallback = null;
+                passwordDialog.open()
+            }
+        }
+
+        MoneroComponents.SettingsListItem {
+            iconText: FontAwesome.key
+            title: qsTr("Show seed & keys") + translationManager.emptyString
```
Comment (rating89us) [inline pages/settings/SettingsWallet.qml]:
```suggestion
            title: qsTr("Show seed & keys (backup)") + translationManager.emptyString
```

File: pages/settings/SettingsWallet.qml (hunk 3 above)
Comment (rating89us) [reply pages/settings/SettingsWallet.qml]: Most items' descriptions describe what is the action of the button. We should keep this consistent.

File: pages/settings/SettingsWallet.qml [hunk 8]
```diff
@@ -114,25 +160,15 @@ Rectangle {
         }
 
         MoneroComponents.SettingsListItem {
-            buttonText: qsTr("Change password") + translationManager.emptyString
-            description: qsTr("Change the password of your wallet.") + translationManager.emptyString
-            title: qsTr("Change wallet password") + translationManager.emptyString
+            iconText: FontAwesome.cashRegister
+            title: qsTr("Enter merchant mode") + translationManager.emptyString
```
Comment (rating89us) [inline pages/settings/SettingsWallet.qml]:
```suggestion
            title: qsTr("Enters merchant mode") + translationManager.emptyString
```

**Comment by selsta** [comment]
@tobtoht ping

Would like to get this merged :)

**Comment by tobtoht** [comment]
@selsta 

Hey, sorry for the delayed response. I've been traveling. I'll get this fixed up this weekend.

**Comment by tobtoht** [comment]
Not changing strings for now as requested by @erciccione .

**Comment by erciccione** [comment]
@tobtoht What are you referring to? i don't remember asking to not make string changes in this PR.

**Comment by tobtoht** [comment]
I may have extrapolated too much, but I was going by: 

https://github.com/monero-project/monero-gui/issues/2660#issuecomment-568169909
https://github.com/monero-project/monero-gui/issues/2653#issuecomment-568168416

**Comment by erciccione** [comment]
@tobtoht I see. My point wasn't to discourage string changes, but only to keep in mind that such changes will require translation efforts. It's important to keep that factor in consideration when evaluating if it worth to change a string or not, but that doesn't mean that we shouldn't make improvements :)

So, if people think the string changes worth the effort necessary to retranslate them and the risk to have them temporarely untranslated in some languages, let's do it.

**Comment by tobtoht** [comment]
@erciccione Okay that's clear, thanks. I want to limit the scope of this PR so we can get it merged in time for the next release. I think string changes can be best discussed in a separate issue if there is a desire to change them.

**Comment by xiphon** [comment]
https://github.com/monero-project/monero-gui/pull/2821

**Comment by selsta** [comment]
#2830 

Thanks again, we split the PR up in two for easier review.

**Pull request closed without merging**